src/avl_tree.jl

```diff
@@ -19,8 +19,9 @@ AVLTreeNode_or_null{T} = Union{AVLTreeNode{T}, Nothing}
 mutable struct AVLTree{T}
     root::AVLTreeNode_or_null{T}
     count::Int
+    lt::Function
```

Comment:
The only thing I am deliberating is the member name, `lt`. I suggest `comp` for better readability of the code. Otherwise this looks good to me. 

Also, try squashing the commits, into single one. A lot of `Update src/avl_tree.jl` doesn't make sense in the long scheme of things. 

The codes looks fine to be merged after these two changes.

Reply:
In isolation I agree, but I went with `lt` for consistency with `sort` and all the other sorting functions, which were the only other Julia functions could think of which take a comparison as an optional parameter. If you still prefer `comp` I can switch it over though.

Reply:
This is an abstractly typed field.
That is going to make things slow as it will cause a dynamic dispatch.

Instead do:
```
mutable struct AVLTree{T, F}
    root::AVLTreeNode_or_null{T}
    count::Int
    lt::F
end
```

A docstring  also needs to be added.